core/types/receipt_opstack.go

```diff
@@ -0,0 +1,48 @@
+package types
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/params"
+)
+
+// deriveOPStackFields derives the OP Stack specific fields for each receipt.
+// It must only be called for blocks with at least two transactions.
```

Comment:
nit: it will still work even if there is only the system deposit transaction right? 
```suggestion
// It must only be called for blocks with at least one transaction (the system deposit).
```

Reply:
I think those transactions don't have receipts though.

I'm generally unsure about this requirement, it does match the condition in the calling function `DeriveFields()`, but I don't understand it. We already skip deposit transactions in the loop below, so I don't know why we can't just call this unconditionally (perhaps it is a performance optimisation?) And what if we had multiple deposits in a block, we would still call this function and potentially skip all the transactions in the loop.

Reply:
> so I don't know why we can't just call this unconditionally (perhaps it is a performance optimisation?)

It would panic since we read `txs[0].Data()`. Ofc we can guard against this by checking the length of the txs slice, but I don't want to bikeshed this comment too much: as long as it accurately documents the behavior I'm happy.

Reply:
Yeah completely empty tx slice would be problematic, but shouldn't happen on an optimism chain, so the tx length check in the caller seems redundant to me still. Doesn't need to block this PR since I think it preserves original semantics, but I would like to either understand this better or circle back and change it.

Reply:
The genesis block might be an edge case.

Reply:
The edge case I wanted to cover with this condition is the Jovian activation block, because it still uses the old L1 attributes deposit. But I realize the old condition doesn't work. I improved the method to test that the last transaction is a non-deposit one at the top and exit early otherwise. This way, the function can also be called if having only a single deposit tx.